### DISCUSSION TITLE
using header/footer adds another margin

### DISCUSSION THREAD

**Discussion by Wuppie007** [gotenberg/gotenberg#434] [Q&A]
I want to add an header and footer to my html page. In this example an black image. But it adds an 'extra margin' at the top and bottom in case of both adding header and footer:

![afbeelding](https://user-images.githubusercontent.com/23653628/157743091-20562448-8262-4090-ae1b-e05004ce5d06.png)

When I remove the set header and set footer commands, this 'extra' margin disappears.

Debug:
running gotenberg version 7.5.0

curl \
--request POST 'http://192.168.x.x:3000/forms/chromium/convert/html' \
--form 'files=@"index.html"' \
--form 'files=@"header.html"' \
--form 'files=@"footer.html"' \
-F 'marginTop="1"' \
-F 'marginBottom="1"' \
-F 'marginLeft="0"' \
-F 'marginRight="0"' \
-o my.pdf

index.html
```<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />   
  </head>
  <body>
    Body
  </body>
</html>
```

header.html and footer.html
``` <html>

<head>
    <meta charset="utf-8">

    <style>
    </style>
</head>

<body>
    <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAA+EAAAB0CAIAAABlmKTcAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAFmSURBVHhe7cEBDQAAAMKg909tDwcEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJyqAUabAAGIM6HnAAAAAElFTkSuQmCC"
         style="max-width:100%;">
</body>
</html>
```

**Comment by gulien** [comment]
Hello @Wuppie007,

AFAIK, these extra margins come from:

```
-F 'marginTop="1"'
-F 'marginBottom="1"'
```

I guess you could play with those form fields' values 😄

**Reply by Wuppie007** [reply]
Unfortunately not, this setting sets te complete margin until the body begins. It's about the header itself. It is about the header itself. So the header has room for lets say 1 'unit'.  But in that space, the contents of the header adds another margin.

I've created a drawing, hoping you are conviced:
![afbeelding](https://user-images.githubusercontent.com/23653628/157841463-335e2dd6-7277-4210-b67d-0e0edac3bea8.png)

**Reply by gulien** [reply]
Oh! Sorry, I misunderstand your issue 😄 . By chance, it seems @idenisovs found a workaround for the same problem: https://github.com/gotenberg/gotenberg/discussions/343#discussioncomment-1315551

**Reply by Wuppie007** [reply]
Thanks for the quick response, the solution work! But if I can contribute to a more future-proof solution, please let me know. Is the caused by this repo or is it caused by a plugin?

**Reply by gulien** [reply]
It's a behavior from Google Chrome, it has nothing to do with Gotenberg alas.
